---
title: ""
author: 'Bernhard Finke'
date: "30/03/2022"
output: html_document
runtime: shiny
pagetitle: "Executable Research Paper: A unified resource and configurable model of the synapse proteome and its role in disease"
---

```{r setup, include=FALSE}
source('AnNetFuncs.R')

library(shiny)
library(dplyr)
library(tidyr)
library(ggplot2)
library(RSQLite)
library(UpSetR)
library(igraph)
library(ggrepel)
library(corrplot)
library(DT)
library(randomcoloR)
library(plotly)



# generate empty plot when selection contains no known proteins
text <- "No proteins in current selection"
EmptyPlot <- ggplot() + annotate("text", x = 4, y = 25, size=8, label = text) +  theme_void()

study.df <- data.frame("Study name"=c("Husi_2000[1]", "Walikonis_2000[2]", "Peng_2004[3]", "Satoh_2002[4]",
                                        "Youshimura_2004[5]", "Farr_2004[6]", "Jordan_2004[7]", "Li_2004[8]",
                                        "Trinidad_2005[9]", "Cheng_2006[10]", "Collins_2006[11]", "Dosemeci_2006[12]",
                                        "Dosemeci_2007[13]", "Trinidad_2008[14]", "Selimi_2009[15]", "Fernandez_2009[16",
                                        "Bayes_2011[17]", "Bayes_2012[18]", "Schwenk_2013[19]", "Distler_2014[20]", 
                                        "Bayes_2014[21]","Uezu_2016[22]", "Focking_2016[23]", "Li_2016[24]", "Fernandez_2017[25]",
                                        "Roy_2017[26]", "Li_2017[27]", "Roy_2018[28]", "Wilson_2019[29]", "Coughenour_2004[30]",
                                        "Blondeau_2004[31]", "Phillips_2005[32]", "Morciano_2005[33]", "Burre_2006[34]",
                                        "Takamori_2006[35]", "Khanna_2007[36]", "Morciano_2009[37]", "Abul-Husn_2009[38]",
                                        "Abul-Husn_2011[39]", "Gorini_2010[40]", "Gronborg_2010[41]", "Boyken_2013[42]",
                                        "Wilhelm_2014[43]", "Brinkmalm_2014[44]", "Weingarten_2014[45]", "Kokotos_2018[46]",
                                        "Taoufiq_2020[47]", "Filiou_2010[48]", "Dahlhaus_2011[49]", "Cohen_2013[50]", 
                                        "Biesemann_2014[51]", "Chang_2015[52]", "Liu_2014[53]", "Distler_2014[20]", 
                                        "Kohansal-Nodehi_2016[54]", "Gonzalez-Lozano[55]", "Alfieri_2017[56]", "Heo_2018[57]",
                                        "Taoufiq_2020[47]"),
                         "Gene_N"=c(77, 29, 237, 45, 436, 73, 393, 139, 236, 289, 620, 114, 276, 2158, 63, 113, 1443, 1552,
                                    34, 3558, 1134, 928, 2021, 1602, 107, 1213, 993, 1071, 2134, 36, 209, 110, 153, 165, 410, 104,
                                    369, 138, 145, 57, 618, 414, 1169, 68, 482, 983, 1466, 2980, 673, 2668, 163, 2077, 1388, 4417,
                                    4961, 1560, 351, 2272, 4439),
                         "Compartment" = c(rep("postsynaptic", times=29), rep("presynaptic", times=18), rep("synaptosome", times=12)),
                         "Brain region" = c("forebrain", "forebrain", "forebrain", "forebrain", "forebrain", "whole brain",
                                            "whole brain", "forebrain", "whole brain", "forebrain and cerebellum", "forebrain",
                                            "hippocampus", "cerebral cortex", "cortex, midbrain, cerebellum, and hippocampus",
                                            "cerebellum", "forebrain", "cortex", "cortex", "whole brain", "hippocampus",
                                            "frontal cortex", "cortex and hippocampus", "whole brain", "hippocampus CA1",
                                            "forebrain", "frontal, parietal, temporal and occipital lobes of the neocortex",
                                            "hippocampus CA1", "frontal, medial and caudal cortex, right caudate putamen, 
                                    right hippocampus, whole hypothalamus, and cerebellum (right half)", "cortex",
                                            "forebrain", "whole brain", "cortex", "whole brain", "whole brain", "cerebral cortex",
                                            "whole brain", "whole brain", "hippocampus and striatum", "striatum", "cortex", 
                                            "cerebral cortex", "cerebral cortex", "cortex and cerebellum", "hippocampus", 
                                            "whole brain", "cerebellum", "whole brain", "whole brain", "visual cortex",
                                            "cortex", "forebrain", "cortex", "hippocampus and prefrontal cortex", "hippocampus",
                                            "cerebellum", "cortex", "telenchephalon", "cortex and hippocampus", "whole brain"),
                         "Species" = c("mouse", "rat", "rat", "mouse", "rat", "rat", "mouse and rat", "rat", "mouse", "rat",
                                       "mouse", "rat", "rat", "mouse", "mouse", "mouse", "human", "mouse", "mouse", "mouse",
                                       "human", "mouse", "human", "mouse", "mouse", "human", "mouse", "mouse", "mouse",
                                       "rat", "rat", "rat", "rat", "rat", "rat", "rat", "rat", "mouse", "rat", "mouse", "rat",
                                       "rat", "rat", "mouse", "mouse", "rat", "rat", "mouse", "mouse", "rat", "mouse", "human",
                                       "mouse", "mouse", "rat", "mouse", "mouse", "mouse", "rat"))

output$studyTable <- renderDataTable({
        datatable(study.df, rownames = FALSE) %>%
            formatStyle('Compartment', target = 'row', backgroundColor = styleEqual(c("postsynaptic", "presynaptic", "synaptosome"),
                                                                                    c('darkgrey', 'lightgrey', "lightgreen"))
            )
    })

con_pre <- dbConnect(drv=RSQLite::SQLite(), dbname="synaptic.proteome_SR_20210408.db.sqlite")
TableDF <- dbGetQuery(conn=con_pre, statement="SELECT HumanEntrez, Paper FROM FullGeneFullPaperFullRegion") %>% unique()
dbDisconnect(con_pre)
TableDF$Paper <- tolower(TableDF$Paper)

GetPapersInSelection <- function(papers){
  papers <- tolower(gsub('\\[.*\\]', '', papers))
  out <- TableDF %>% filter(Paper %in% papers)
  out$Paper <- paste(toupper(substr(out$Paper, 1, 1)), substr(out$Paper, 2, nchar(out$Paper)), sep="")
  return(out)
}

output$downloadTable1 <- downloadHandler(
    filename = function() {
      "tableproteins.csv"
    },
    content = function(file) {
      write.csv(GetPapersInSelection(input$tablepapers), file, row.names = FALSE)
    }
  )

output$fig1png <- renderImage({
        list(src = 'figures/fig1.png', height=600)}, deleteFile = FALSE
    )

output$fig2png <- renderImage({
        list(src = 'figures/fig2.png', height=500)}, deleteFile = FALSE
    )

output$fig3png <- renderImage({
        list(src = 'figures/fig3.png', height=570)}, deleteFile = FALSE
    )

selectClustering <- function(id, x){
    # selector for clustering method
    selectInput(
        inputId = paste("Cluster", id, sep=""),
        label = "Clustering method",
        choices = c("Louvain", "Infomap", "Leiden"),
        selected = "Louvain",
        multiple = FALSE
    )
}


selectLocalisationNoAll <- function(id, multi=FALSE, select="Postsynaptic"){
    # selector for Localisation (does not include option All)
    selectInput(
        inputId = paste("LocalisationnoAll", id, sep=""),
        label = "Sub-cellular Localisation",
        choices =  c("Postsynaptic" = "post", "Presynaptic" = "pre", "Synaptosome" = "syn"),
        selected = select,
        multiple = multi
    )
}



knitr::opts_chunk$set(echo = TRUE)

```


```{css, echo=FALSE}

.figs {
  align: center;
  display:block;
  padding:9.5px;
  margin:0 0 2px;
  margin-top:2px;
  line-height:20px;
  border:3px solid rgba(0,0,0,0.15);
  border-radius:4px; 
}
h1,h2,h3,h4,h5 {
  line-height: 25px;
  }    




```

```{r intro, echo=FALSE}

titlePanel(HTML("Executable Research Paper: <a href='https://www.nature.com/articles/s41598-021-88945-7'>A unified resource and configurable model of the synapse proteome and its role in disease</a>"))
        
h2("Original Paper Authors:")
        
h4("Oksana Sorokina, Colin Mclean, Mike D. R. Croning, Katharina F. Heil, Emilia Wysocka, Xin He, David Sterratt, Seth G. N. Grant, T. Ian Simpson, J. Douglas Armstrong")
        
h2("Executable Paper Author:")
        
h4("Bernhard Finke")
        
h1("Abstract")
        
h4("Genes encoding synaptic proteins are highly associated with neuronal disorders many of which show clinical co-morbidity. We integrated
    58 published synaptic proteomic datasets that describe over 8000 proteins and combined them with direct protein-protein interactions and
    functional metadata to build a network resource that reveals the shared and unique protein components that underpin multiple disorders. 
    All the data are provided in a flexible and accessible format to encourage custom use.")

h1("Introduction")
        
h4("At neuronal synapses, the proteomes in presynaptic and postsynaptic compartments form complex and highly dynamic molecular networks.
    These networks mediate signal transduction and plasticity processes that underpin normal (and abnormal) information processing in the brain.
    We systematically curated proteomic datasets dating from 2000 to 2020, to produce a comprehensive index of the proteins (and their genes) 
    expressed at the mammalian synapse (see Methods for details). This resulted in 58 papers, which when combined, describe a landscape of 8087 
    synaptic genes.")
        
h4(HTML("The set includes 29 post synaptic proteome (PSP) studies (2000 to 2019) contributing a total of 5560 mouse and human unique gene identifiers;
         18 presynaptic studies (2004 to 2020) describe 2772 unique human and mouse gene IDs, and 11 studies that span the whole synaptosome and report 
         7198 unique genes (<a href='#studyTable'>Table 1</a>, <a href='https://www.nature.com/articles/s41598-021-88945-7#MOESM4'>Supplementary Table           1</a>)."))
        
h3(HTML("<b>Table 1</b>: Studies included in the database. Dark grey corresponds to postsynaptic, light grey-to presynaptic, and green-to synaptosomal          studies:"))

inputPanel(
  selectInput(
    inputId = "tablepapers",
    label = "Download proteins found in selected papers",
    choices =  study.df$Study.name,
    selected = "Husi_2000[1]",
    multiple = TRUE
  ),
  downloadButton("downloadTable1")
)
        
fluidRow(
        column(12,
        dataTableOutput('studyTable')
        ))
        
h4("Each study was annotated with relevant metadata including GO function, disease association and cross-ref to SynGo. Orthologues were mapped 
    across human, mouse and rat and each mapped onto stable identifiers (MGI, Entrez and Uniprot).")
        
h4(HTML("High throughput proteomic techniques are powerful, but they are noisy, and contamination is always a concern. A large number (2091 for PSP 
       and 1434 for presynapse, Fig. 1A,B) of proteins have been observed just once. While single hits may be accounted for lack        of sensitivity with low abundance molecules, it could also indicate the presence of false positive components brought in by experimental                uncertainty."))

div(class="figs", 
            
h1(HTML("<b>Figure 1</b>")),
        
imageOutput('fig1png', height=650),
        
h5(HTML('<b>(A)</b> Discovery rate of new PSD proteins across 29 postsynaptic studies, where the number of proteins is plotted against the frequency of identification. 2091 PSP proteins have been observed just once. The most frequently found proteins (i.e. <br/> detected in 22, or more, studies 
       out of the 29) include very well-known PSD proteins, for example: DLG4 (28/29), CAMK2A (27/29), INA (26/29), SPTBN1, CAMK2B, 
       DLG2, NSF, GRIN2B, GRIN1 (25/29), BIAP2, BSN (24/29) (full list in <br/> <a href="https://www.nature.com/articles/s41598-021-88945-7#MOESM5">Supplementary Table 2</a>). <b>(B)</b> Discovery rate of new proteins analysed 
       across 18 presynaptic studies. More than half of the proteins in the presynaptic proteome (1251) have been observed just once.The most 
       frequent presynaptic genes include AP2B1, HSPA8, GNAO1, ACTB (15/17), STX1B, ATP6V0A1, STXBP1, ATP1A3, ATP6V1E1, SYT1, 
       GNB1, TUBA1A, <br/> VAMP2, NSF, DNM1 (14/17) with full statistics available in <a                           href="https://www.nature.com/articles/s41598-021-88945-7#MOESM6">Supplementary Table 3</a>. <b>(C)</b> Contribution of each of 29
       studies to <br/> the total number of PSP genes (purple-total number of genes, yellow-identified in this study). Two major jumps in the gross 
       number of proteins identified occur in 2008, when 1249 new proteins were reported by [14] and in 2014 with 2588 new <br/> proteins added by 
       [20]. <b>(D)</b> Contribution of each of 18 studies to the total number of presynaptic genes (purple-total number <br/> of genes,                      yellow-identified in this study): two jumps in newly discovered proteins correspond to studies in years 2010 and <br/> 2014. <b>(E)</b>                Accumulation of the new  PSP genes (black) compared to the total datasets (blue) over years. <b>(F)</b> Accumulation of new presynaptic genes          (black) compared to the total datasets (blue) over years. <b>(G)</b> Non-linear fit predicting the total size of "consensus" PSP (genes found in        two and more studies, 3499) (P = 2.36E-11, residual standard error: 192.7 on 12 degrees <br/> of freedom) by year 2023 which, when compared to         the current number (3438) indicates that our knowledge on PSP components, based on currently available methodologies, is close to saturation.          <b>(H)</b> Overlap of three synaptic datasets: presynaptic, postsynaptic and synaptosomal. Bars correspond to the number of unique genes in each        compartment and their intersections.'))
)

h1("Executable Implementations of Figure 1")
        
h4("This section contains executable and interactive versions of the plots contained above in figure 1. Selectors are labeled to indicate which selection will give the exact version contained in the original paper.")

```

```{r titledb, echo=FALSE}
h5("Code used to connect to SQLite database and define necessary data frame for analysis in figure 1:")
```


```{r load.db, echo=TRUE}
con <- dbConnect(drv=RSQLite::SQLite(), dbname="synaptic.proteome_SR_20210408.db.sqlite")


GeneDFraw <- dbGetQuery(conn=con, statement="SELECT Localisation, BrainRegion, MethodID,
                                          HumanEntrez, Paper, Year FROM FullGeneFullPaperFullRegion")

Methods <- dbGetQuery(conn=con, statement="SELECT ID, Name FROM Method")

GeneDF <- as.data.frame(GeneDFraw) %>% merge(Methods, by.x="MethodID", by.y="ID", all=TRUE) %>% 
  select(-MethodID) %>% rename('Method' = 'Name')

dbDisconnect(con)
```

```{r titlehelper1, echo=FALSE}
h5("Defines useful helper functions required for subsetting data:")
```


```{r fig1.helper, echo=TRUE}
# get all genes with a given localisation, brain region, and sequencing method
GenerateSubset <- function(loc, reg, met){
    subset <- data.frame(GeneDF)
    if (loc != "All"){
        subset <- subset %>% filter(Localisation == loc)
    }
    if (reg != "All"){
        subset <- subset %>% filter(BrainRegion == reg)
    }
    if (met != "All"){
        subset <- subset %>% filter(Method == met)
    }
    return(subset)
}

# determine number of papers all genes in a specific subset have been identified in
GetNumPapersSubset <- function(loc, reg, met){
    subset <- GenerateSubset(loc, reg, met)
    PaperDF <- subset %>% group_by(HumanEntrez) %>% summarise(NumPapers = length(unique(Paper)))
    return(merge(subset, PaperDF, by.x="HumanEntrez", by.y="HumanEntrez", all=TRUE))
}


# get all genes with a given localisation, brain region, and sequencing method and identified in at least numpap papers
SubsetNumPapers <- function(loc, reg, met, numpap){
    subset <- GetNumPapersSubset(loc, reg, met)
    return(subset[subset$NumPapers >= as.numeric(numpap), ])
}


```

```{r title1a1b, echo=FALSE}
h3("Fig 1a and 1b: Discovery rate of new proteins across all studies, \n where the number of proteins is plotted against the frequency of identification")
```

```{r fig1a1b}
GenerateFig1a1b <- function(loc, reg, met){
  df <- GetNumPapersSubset(loc, reg, met)
  first_occurrence <- df[match(unique(df$HumanEntrez), df$HumanEntrez),]
  dfOut <- as.data.frame(table(first_occurrence$NumPapers)) 
  if (nrow(dfOut) != 0){
      dfOut <- dfOut %>% rename('NumGenes' = 'Var1')
  }
  return(dfOut)
}

fig1a1b <- reactive({
  data1a1b  <- GenerateFig1a1b(input$Localisation1, input$BrainRegion1, input$Method1)
  if (nrow(data1a1b) != 0) {
      ggplot(data1a1b, aes(x=NumGenes, y=Freq)) + geom_bar(stat="identity") + 
      xlab("Number of Protein Identifications") + ylab("Number of Proteins")
  }
  else {
      EmptyPlot
  }
})

```



```{r 1a1boutput, echo=FALSE}
output$download1a1b <- downloadHandler(
  filename = function(){"data1a1b.csv"},
  content = function(file){write.csv(GenerateFig1a1b(input$Localisation1, input$BrainRegion1, input$Method1), file, row.names = FALSE)}
  )


inputPanel(
  selectInput(
    inputId = "Localisation1",
    label = "Sub-cellular Localisation",
    choices =  c("Postsynaptic (Fig 1a)" = "Postsynaptic","Presynaptic   (Fig 1b)" = "Presynaptic", "Synaptosome", "All"),
    selected = "Postsynaptic",
    multiple = FALSE
  ),
  selectInput(
    inputId = "BrainRegion1",
    label = "Brain Region",
    choices = c("All (Fig 1a & 1b)" = "All", "Brain", "Forebrain", "Midbrain", "Cerebellum", "Hypothalamus", "Hippocampus", "Striatum", "Cerebral cortex", "Frontal lobe", "Occipital lobe", "Temporal lobe", "Parietal Lobe", "Telencephalon", "Prefrontal cortex", "Motor cortex", "Visual cortex"), 
    selected = "All",
    multiple = FALSE
  ),
  selectInput(
    inputId = "Method1",
    label = "Method",
    choices = c("Target", "Shotgun", "All (Fig 1a & 1b)" = "All"), 
    selected = "All",
    multiple = FALSE
  ),
  downloadButton("download1a1b", "Download current dataset")
)

renderPlot(fig1a1b())


```

```{r title1c1d, echo=FALSE}
h3("Fig 1c and 1d:  Contribution of each study \n to the total number of genes in each category")
```


```{r fig1c1d}
GenerateFig1c1d <- function(loc, reg, met, numpap){
  df <- SubsetNumPapers(loc, reg, met, numpap)
  total <- length(unique(df$HumanEntrez))
  df <- df %>% group_by(Paper, Year) %>% summarise(NumGenes = length(unique(HumanEntrez)), .groups="keep") %>% 
      mutate(NumGenes = NumGenes / total)
  df$stat <- "Found"
  dfNot <- df %>% select("Paper", "Year", "NumGenes")
  dfNot <- dfNot %>% mutate(NumGenes = 1 - NumGenes)
  dfNot$stat <- "NotFound"
  dffull <- rbind(dfNot, df)
  if (nrow(dffull) != 0){
      dffull$stat <- relevel(as.factor(dffull$stat), "NotFound")    
  }
  return(dffull)
}


fig1c1d <- reactive({
  data1c1d  <- GenerateFig1c1d(input$Localisation2, input$BrainRegion2, input$Method2, input$NumPapers2)
  if (nrow(data1c1d) != 0) {
      ggplot(data1c1d, aes(x=reorder(Paper, Year), y=NumGenes, fill= stat)) + geom_bar(stat="identity")  + 
      ylab("count") + theme(axis.text.x=element_text(angle=90)) + xlab("") + 
      scale_fill_manual(values=c("darkorchid4", "yellow"))
  }
  else {
      EmptyPlot
  }
})
```


```{r 1c1doutput, echo=FALSE}
output$download1c1d <- downloadHandler(
  filename = function(){"data1c1d.csv"},
  content = function(file){write.csv(GenerateFig1c1d(input$Localisation2, input$BrainRegion2, input$Method2, input$NumPapers2), file, row.names = FALSE)}
  )

inputPanel(
  selectInput(
    inputId = "Localisation2",
    label = "Sub-cellular Localisation",
    choices =  c("Postsynaptic (Fig 1c)" = "Postsynaptic", "Presynaptic   (Fig 1d)" = "Presynaptic", "Synaptosome", "All"),
    selected = "Postsynaptic",
    multiple = FALSE
  ),
  selectInput(
    inputId = "BrainRegion2",
    label = "Brain Region",
    choices = c("All (Fig 1c & 1d)" = "All", "Brain", "Forebrain", "Midbrain", "Cerebellum", "Hypothalamus", "Hippocampus", "Striatum", "Cerebral cortex", "Frontal lobe", "Occipital lobe", "Temporal lobe", "Parietal Lobe", "Telencephalon", "Prefrontal cortex", "Motor cortex", "Visual cortex"), 
    selected = "All",
    multiple = FALSE
  ),
  selectInput(
    inputId = "Method2",
    label = "Method",
    choices = c("Target", "Shotgun", "All (Fig 1c & 1d)" = "All"), 
    selected = "All",
    multiple = FALSE
  ),
  selectInput(
    inputId = "NumPapers2",
    label = "Evidence (in >= n papers)",
    choices = c("1 (Fig 1c & 1d)" = 1, 2, 3, 4, 5, 6, 7, 8, 9, 10), 
    selected = 1,
    multiple = FALSE
  ),
  downloadButton("download1c1d", "Download current dataset")
)

renderPlot(fig1c1d())


```

```{r title1e1f, echo=FALSE}
h3("Fig 1e and 1f: Accumulation of new genes (black) \n compared to the total datasets (blue) over years")
```


```{r fig1e1f}
GenerateFig1e1f <- function(loc, reg, met, numpap){
  df <- SubsetNumPapers(loc, reg, met, numpap) %>% arrange(Year)
  if (nrow(df) != 0){
      df <- df[match(unique(df$HumanEntrez), df$HumanEntrez),] %>% group_by(Year) %>%
        summarise(GenesInYear = length(HumanEntrez)) %>%
          complete(Year = 2000:2020, fill = list(GenesInYear = 0))
      df$TotalGenes <- cumsum(df$GenesInYear)
  }
  return(df)
}


fig1e1f <- reactive({
  data1e1f  <- GenerateFig1e1f(input$Localisation3, input$BrainRegion3, input$Method3, input$NumPapers3)
  if (nrow(data1e1f) != 0) {
      ggplot(data1e1f, aes(x=Year, y=TotalGenes)) + geom_area(fill="#56B4E9") +
      geom_area(aes(x=Year, y=GenesInYear)) + xlab("years") + ylab("total")
  }
  else {
      EmptyPlot
  }
})
```


```{r 1e1foutput, echo=FALSE}
output$download1e1f <- downloadHandler(
  filename = function(){"data1e1f.csv"},
  content = function(file){write.csv(GenerateFig1e1f(input$Localisation3, input$BrainRegion3, input$Method3, input$NumPapers3), file, row.names = FALSE)}
  )

inputPanel(
  selectInput(
    inputId = "Localisation3",
    label = "Sub-cellular Localisation",
    choices =  c("Postsynaptic (Fig 1e)" = "Postsynaptic", "Presynaptic   (Fig 1f)" = "Presynaptic", "Synaptosome", "All"),
    selected = "Postsynaptic",
    multiple = FALSE
  ),
  selectInput(
    inputId = "BrainRegion3",
    label = "Brain Region",
    choices = c("All (Fig 1e & 1f)" = "All", "Brain", "Forebrain", "Midbrain", "Cerebellum", "Hypothalamus", "Hippocampus", "Striatum", "Cerebral cortex", "Frontal lobe", "Occipital lobe", "Temporal lobe", "Parietal Lobe", "Telencephalon", "Prefrontal cortex", "Motor cortex", "Visual cortex"), 
    selected = "All",
    multiple = FALSE
  ),
  selectInput(
    inputId = "Method3",
    label = "Method",
    choices = c("Target", "Shotgun", "All (Fig 1e & 1f)" = "All"), 
    selected = "All",
    multiple = FALSE
  ),
  selectInput(
    inputId = "NumPapers3",
    label = "Evidence (in >= n papers)",
    choices = c("1 (Fig 1e & 1f)" = 1, 2, 3, 4, 5, 6, 7, 8, 9, 10), 
    selected = 1,
    multiple = FALSE
  ),
  downloadButton("download1e1f", "Download current dataset")
)

renderPlot(fig1e1f())


```

```{r title1g, echo=FALSE}
h3("Fig 1g: Non-linear fit predicting the total size of consensus genes")
```


```{r fig1g}
fig1g <- reactive({
  data1g <- GenerateFig1e1f(input$Localisation4, input$BrainRegion4, input$Method4, input$NumPapers4)
  if (nrow(data1g) != 0) {
      if (input$fitType == "Linear"){
          fit <- geom_smooth(method='lm', formula= y~x, fullrange = TRUE, se = FALSE)
          error <- round(summary(lm(Year~TotalGenes, data=data1g))$r.squared, 2)
          ggplot(data1g, aes(x=Year, y=TotalGenes)) + geom_point() + fit + xlab("years") + 
              ylab("total") + geom_text(label = paste("R-squared:", error), x = 2015, y = 150, size = 5)
      } 
      else if (input$fitType == "Logistic"){
          check <- 0
          for (x in data1g$GenesInYear){
              if (x != 0){
                  check <- check + 1
              }
              if (check > 1){
                  break
              }
          }
          if (check == 1){
              text <- "Only one year in current selection: \n no logistic regression can be fitted"
              ggplot() + annotate("text", x = 4, y = 25, size=8, label = text) +  theme_void()
          }
          else {
              fit <- geom_smooth(method='nls', formula= y ~ SSlogis(x, Asym, xmid, scal), fullrange = TRUE, 
                                 se = FALSE)
              asymptote <- round(coef(nls(TotalGenes ~ SSlogis(Year, Asym, xmid, scal), data = data1g))[1])
              ggplot(data1g, aes(x=Year, y=TotalGenes)) + geom_point() + fit + 
                  xlab("years") + ylab("total") + xlim(2000, 2050) + geom_hline(yintercept = asymptote) +
                  geom_text(label = paste("Total predicted \n genes:", asymptote), x = 2040, y = asymptote / 2, 
                            size = 5)
          }
      }
      
  }
  else {
      EmptyPlot 
  }
})
```


```{r 1goutput, echo=FALSE}
output$download1g <- downloadHandler(
  filename = function(){"data1g.csv"},
  content = function(file){write.csv(GenerateFig1e1f(input$Localisation4, input$BrainRegion4, input$Method4, input$NumPapers4), file, row.names = FALSE)}
  )

inputPanel(
  selectInput(
      inputId = "Localisation4",
      label = "Sub-cellular Localisation",
      choices =  c("Postsynaptic (Fig 1g)" = "Postsynaptic", "Presynaptic", "Synaptosome", "All"),
      selected = "Postsynaptic",
      multiple = FALSE
  ),
  selectInput(
      inputId = "BrainRegion4",
      label = "Brain Region",
      choices = c("All (Fig 1g)" = "All", "Brain", "Forebrain", "Midbrain", "Cerebellum", "Hypothalamus", "Hippocampus", "Striatum", "Cerebral cortex", "Frontal lobe", "Occipital lobe", "Temporal lobe", "Parietal Lobe", "Telencephalon", "Prefrontal cortex", "Motor cortex", "Visual cortex"), 
      selected = "All",
      multiple = FALSE
  ),
  selectInput(
      inputId = "Method4",
      label = "Method",
      choices = c("Target", "Shotgun", "All (Fig 1g)" = "All"), 
      selected = "All",
      multiple = FALSE
  ),
  selectInput(
      inputId = "NumPapers4",
      label = "Evidence (in >= n papers)",
      choices = c(1, "2 (Fig 1g)" = 2, 3, 4, 5, 6, 7, 8, 9, 10), 
      selected = 2,
      multiple = FALSE
  ),
  selectInput(
      inputId = "fitType",
      label = "Type of Fit",
      choices = c("Logistic (Fig 1g)" = "Logistic", "Linear"), 
      selected = "Logistic",
      multiple = FALSE
  ),
  downloadButton("download1g", "Download current dataset")
)

renderPlot(fig1g())


```


```{r title1h, echo=FALSE}
h3("Fig 1h: Overlap of three synaptic datasets: presynaptic, \n postsynaptic and synaptosomal. Bars correspond to the number of unique genes in each  \n compartment and their intersections")
```


```{r fig1h}
GenerateFig1h <- function(reg, met, numpap){
  df <- SubsetNumPapers("All", reg, met, numpap) %>% select(HumanEntrez, Localisation)
  df <- unique(df[,c('HumanEntrez','Localisation')])
  if (nrow(df) != 0){
      setPost <- df %>% filter(Localisation == "Postsynaptic") %>% select(-Localisation)
      setPre <- df %>% filter(Localisation == "Presynaptic") %>% select(-Localisation)
      setSynap <- df %>% filter(Localisation == "Synaptosome") %>% select(-Localisation)
      setAll <- intersect(intersect(setPost, setSynap), setPre)
      setonlyPre <- setdiff(setdiff(setPre, setPost), setSynap)
      setonlyPost <- setdiff(setdiff(setPost, setPre), setSynap)
      setonlySynap <- setdiff(setdiff(setSynap, setPre), setPost)
      setSynapPost <- setdiff(intersect(setPost, setSynap), setPre)
      setPostPre <- setdiff(intersect(setPre, setPost), setSynap)
      setSynapPre <- setdiff(intersect(setSynap, setPre), setPost)
      output <- c("syn" = nrow(setonlySynap), "psd" = nrow(setonlyPost), "pres" = nrow(setonlyPre), 
                  "syn&psd" = nrow(setSynapPost), "syn&pres" = nrow(setSynapPre), 
                  "psd&pres" = nrow(setPostPre), "psd&pres&syn" = nrow(setAll))
      check <- 0
      for (x in output){
          if (x != 0){
              check <- check + 1
          }
          if (check > 1){
              break
          }
      }
      if (check == 1){
          output <- c(1, 1)
      }
  }
  else {
      output <- 1
  }
  return(output)
}


fig1h <- reactive({
  data1h <- GenerateFig1h(input$BrainRegion5, input$Method5, input$NumPapers5)
  if (length(data1h) == 7) {
      upset(fromExpression(data1h), text.scale=1.5, mainbar.y.label = "Number of genes", 
            sets.x.label = "Compartments", sets.bar.color = c("purple4", "#56B4E9", "orangered3"))
  }
  else if (length(data1h) == 1) {
      EmptyPlot
  }
  else{
      text <- "No overlap: current selection contains \n proteins uniquely found in one compartment"
      ggplot() + annotate("text", x = 4, y = 25, size=8, label = text) +  theme_void()
  }
})
```


```{r 1houtput, echo=FALSE}
output$download1h <- downloadHandler(
  filename = function(){"data1h.csv"},
  content = function(file){write.csv(GenerateFig1h(input$BrainRegion5, input$Method5, input$NumPapers5), file, row.names = TRUE)}
  )

inputPanel(
  selectInput(
    inputId = "BrainRegion5",
    label = "Brain Region",
    choices = c("All (Fig 1h)" = "All", "Brain", "Forebrain", "Midbrain", "Cerebellum", "Hypothalamus", "Hippocampus", "Striatum", "Cerebral cortex", "Frontal lobe", "Occipital lobe", "Temporal lobe", "Parietal Lobe", "Telencephalon", "Prefrontal cortex", "Motor cortex", "Visual cortex"), 
    selected = "All",
    multiple = FALSE
  ),
  selectInput(
    inputId = "Method5",
    label = "Method",
    choices = c("Target", "Shotgun", "All (Fig 1h)" = "All"), 
    selected = "All",
    multiple = FALSE
  ),
  selectInput(
    inputId = "NumPapers5",
    label = "Evidence (in >= n papers)",
    choices = c("1 (Fig 1h)" = 1, 2, 3, 4, 5, 6, 7, 8, 9, 10), 
    selected = 1,
    multiple = FALSE
  ),
  downloadButton("download1h", "Download current dataset")
)

renderPlot(fig1h())


```


```{r fig2, echo=FALSE}
h4(HTML('The rate of growth with respect to newly discovered proteins for PSP appears to be slowing (Fig. 1C,E) and therefore there is now an opportunity to define a more reliable subset. Following the approach described in [11], we selected genes found in two or more independent studies to designate the "consensus" PSP. This resulted in 3,438 genes, which is ~ 7 times larger than reported by [11] and described a subset of synaptic proteins for which have higher confidence. In this subset we observe the increment of new genes per year decreases after 2008 and drops completely after 2014 (Fig. 1C). Based on this, we predict a total number of consensus PSP genes found to be 3499 (Fig. 1G) by year 2023 which, when compared to the current number indicates that our knowledge on PSP components, based on currently available methodologies, is close to saturation.'))
        
h4(HTML('It is different for the presynaptic compartment, where the recent trend in newly identified genes indicates that saturation has not been achieved yet (Fig. 1D,F). For instance, the latest study by Taoufiq et al. [47] brought in over 400 new genes to our presynaptic list.'))
        
h4(HTML('The overlap of proteins found in pre- and post-synaptic datasets, and proteins identified in synaptosomal studies is shown at Fig. 1H and <a href="https://www.nature.com/articles/s41598-021-88945-7#MOESM3">Fig. 1 in Supplementary Methods.</a>'))
        
h4('To reconstruct protein-protein interaction (PPI) networks for the pre- and post-synaptic proteomes we used human PPI data filtered for the highest confidence direct andphysical interactions from BioGRID [58], Intact [59] and DIP [60]. The resulting PSP network contains 4817 nodes and 27,788 edges in the Largest Connected Component (LCC). The presynaptic network is significantly smaller and comprises 2221 nodes and 8678 edges in the LCC.')
        
h4(HTML('The resulting network model is embedded into a SQLite implementation allowing users to derive custom network models based on meta-data including species, disease association, synaptic compartment, brain region, and method of extraction (Fig. 2). The database with manual is available from <a href="https://www.nature.com/articles/s41598-021-88945-7#MOESM7">Supplementary Materials</a> and from <a href="https://doi.org/10.7488/ds/3017">Edinburgh DataShare</a>, along with a SQLite Studio manual and Rmd file for querying under the R environment, a screencast walk-through demonstrating use-cases can also be found <a href="https://youtu.be/oaW9Yr9AkXM">here</a>.'))

div(class="figs",
            
h1("Figure 2"), 
            
imageOutput('fig2png', height=500), 
            
h5(HTML("Structure of the SQLite database, which includes 58 synaptic studies covering 8087 unique genes and 407,643 direct protein interactions. Grey 
    ovals on the top show the annotated metadata: left-for nodes/ genes, which include brain region, subcellular compartment, method of 
    extraction, disease and GO function annotation and link to published quantitative models; right-for edges/PPIs, which include PSI-MI type 
    and method. The orange ovals in the bottom illustrate the possible outcomes of the database, including: (1) information for specific 
    protein/gene, and (2) information that could be obtained from PPI network, e.g., protein's topological importance, community to disease 
    relationship, and disease-disease comorbidity. The database is available as a <a href='https://www.nature.com/articles/s41598-021-88945-7#MOESM1'>Supplementary File</a> and from <a href='https://doi.org/10.7488/ds/3017'>Edinburgh DataShare.</a>"))
)

h4('The dataset can be used to answer frequent questions such as "What is known about my favourite gene? Is it pre- or postsynaptic? Which brain region was it identified in?". Beyond that, users can extend these queries to extract custom networks based on bespoke subsets of molecules. Worked examples that are easy to customise are shown in the Supplementary files.')
        
h4(HTML("The underlying principle of a systems biology approach is that structural features (pathways and subnetworks) underpin network functionality and given a network, one should be able to extract these features. Clustering algorithms [61,62] are commonly used to identify local communities within the network under the assumption that shared network topology correlates with shared function (and dysfunction). However, the more important question is how the different communities are organised to enable a controllable flow of signals across the large network. Using the PSP network as example, we identified 1029 \"Bridging\" proteins as those known to interact locally with neighbours in the network-helping organise function inside communities they belong to [63,64], and simultaneously influence other communities in the network (Fig. 3A, Methods). Using graph entropy as a compliment means of ranking a protein's ability to inhibit or enhance information flow [65], we found that proteins with high Bridgeness value have ability to decrease the entropy of the network thus facilitating the signal transmission (Fig. 3B,D, Methods). Of the 1029 candidate Bridging proteins (see Region 1, Fig. 3C), we found ~ 43%) associated with at least one known synaptopathy and ~ 21% linked to multiple diseases including: APP (AD&Epi&ASD&PD&HTN&MS&FTD), VDAC1 (AD&PD&MS), and MAPK14 (AD&SCH&HD&HTN&MS), which supports the functional/disease importance of \"bridging\" proteins. Indeed, we found significant overrepresentation for specific diseases such as AD (P = 3.4E-6), HTN (P = 2.1E-5), HD (P = 5.2E-5), PD (P = 2.6E-3) (<a href='https://www.nature.com/articles/s41598-021-88945-7#MOESM5'>Supplementary Table 2</a>)."))

div(class="figs", 
            
h1("Figure 3"), 
            
imageOutput('fig3png', height=600), 
            
h5(HTML("<b>(A)</b> Community structure of the PSP network using the Spectral modularity method. Communities are coloured using the average gene-
    community probability values: bluer coloured a community is, the more probable the genes are of belonging to that community on average. 
    Nodes coloured magenta highlight the core PSD95 interactors [25], which is also highlighted magenta in the Bridgeness plot in <b>(C)</b>. 
    <b>(B)</b> Graph entropy plots: (main) Global graph entropy rate (SR) plot comparing the structure of the PSP network (0.668) against 1000 
    randomised Erdos-Renyi (E-R = 0.989 + - 0.0005) and Power-Law (P-L = 0.9127 + - 0.0032, \u03B1PSP = 2.41) models of similar size, (Enlarged) 
    Evidence for scale-free structure in PSP network using a perturbation analysis (10], plotted is the SR values after each protein is perturbed 
    through over-expression (SR_UP = red) and under-expression (SR_OWN = green), against the log of the proteins degree,. <b>(C)</b> Bridging 
    proteins, estimated using the Spectral clustering algorithm are plotted against semi-local centrality (Methods), allowing their categorisation: 
    Region 1, proteins having a 'global' rather than 'local' influence in the network (also been called bottle-neck bridges, connector or kinless 
    hubs [12] (DLG4, GRIN2B, CAMK2A, etc.). Region 2, proteins having 'global' and 'local' influence (EGFR, HRAS, NRAS, etc.). Region 3, 
    proteins centred within the community they belong to, but also communicating with a few other specific communities (GRIN1, GRIA2-4). 
    Region 4, proteins with 'local' impact , primarily within one or two communities (local or party hubs [9]. <b>(D)</b> Correlation plot for 
    different centrality measures estimated for PSP network.: SP - a protein's shortest path value, SR_UP-Entropy rate when protein is over 
    expressed, SR_DOWN-entropy rate when protein is under expressed, COUNT - number of protein identifications in the studies, Bet - 
    protein's betweenness centrality value, Degree-protein degree, PR- Page Rank, BRIDGESpectral -protein Bridgeness value, CNorm - 
    Protein's local centrality value, Closeness - protein's closeness value; correlation between SR_UP and Bridgeness indicates that genes with 
    higher Bridgeness values also lower the graphs entropy when active/overexpressed, which allows the signal to pass more freely 
    (<a href='https://www.nature.com/articles/s41598-021-88945-7#MOESM5'>Supplementary Table 2</a>). <b>(E)</b> left: Disease-disease relationship for      presynaptic (red) and PSD full (blue) and PSD consensus (green) 
    interactome. Where significance q-values < 0.05 is delineated by the dashed line. Schizophrenia (SCH), Autistic Spectrum Disorder (ASD),
    Autistic Disorder (AUT), Bipolar Disorder (BD), Intellectual Disability (ID), Alzheimer disease (AD), Epilepsy Syndrome (Epi), 
    Parkinson's Disease (PD), Frontotemporal Dementia (FTD), Huntington's Disease (HD) and Multiple Sclerosis (MS) are considered; right: 
    randomisation studies for disease-disease pairs overlap, yellow arrow shows the measured value of Z-score compared to 10,000 AD-HTN, 
    PD-HTN and AD-PD random models. <b>(F)</b> Colocalization of AD and HTN on the PSP network by propagating these gene-disease 
    associations (GDA) through the network using the Belief Propagation DC-SBM algorithm [13]. The colocalization of AD and HTN shared 
    common molecular pathways in communities 31 and 43, which were also found enriched for axon guidance, stress-activated MAPK cascade 
    and response to oxidative stress GO BP terms.")))

h1("Executable Implementations of Figure 3")
        
h4("This section contains executable and interactive versions of the plots contained above in figure 3. Some differences exist between the figures in the paper and the executable ones found here. One reason for this is that the graph visualisation platform 'Gephi' was used to create many of the figures in the original paper, while this implementation uses only R. For figures 3c and 3f, the original clustering algorithm used was Spectral clustering, which is not available in this executable paper. In figure 3c, the highlighted core genes were hand-picked in the original paper, while here they represent genes found in almost all of the papers. In figure 3d, different centrality measures are given.  In figure 3e, postsynaptic consensus genes are not available. Figure 3f does not allow for annotating GO biological processes. Additionally, the authors of the original paper deemed figure 3a to be infeasible to include in the executable paper, while figure 3b is not included as replicating the figure was determined to be superfluous. Additionally, as figure 3c can take up to 30 seconds to load, it does not auto-load with the webpage, but can be loaded by the user using the supplied button.")

h4(HTML("The analysis in these figures was made possible by the 'AnNet' package, developed by Anatoly Sorokin, Colin Mclean, and Oksana Sorokina. It can be found on Github using the following link: <a href='https://github.com/lptolik/AnNet'>https://github.com/lptolik/AnNet</a>"))

```


```{r title3c, echo=FALSE}
h3("Fig 3c: Bridging proteins, estimated using a given clustering \n algorithm are plotted against a given centrality measure")
```


```{r fig3c}

# cluster graphs were saved as recalculating them from scratch takes too long to function in an executable paper
pre_cluster <- read.graph("clusters/pre_cluster.gml", format=c("gml"))
post_cluster <- read.graph("clusters/post_cluster.gml", format=c("gml"))
syn_cluster <- read.graph("clusters/syn_cluster.gml", format=c("gml"))

# to calculate these graphs from scratch please install the 'AnNet' package and follow these steps (example given for presynaptic graph):

# cid<-match('Presynaptic',getCompartments()$Name) 
# t<-getAllGenes4Compartment(cid)
# gg<-buildFromSynaptomeByEntrez(t$HumanEntrez)
# gg<-annotateGeneNames(gg)
# afile<-system.file("extdata", "flatfile_human_gene2HDO.csv", package = "AnNet")
# dis<- read.table(afile,sep="\t",skip=1,header=F,strip.white=T,quote="")
# gg<-annotate_topOnto_ovg(gg, dis)
# gg <- calcCentrality(gg)
# gg <- calcAllClustering(gg)


# besides the graph files we also saved the consensus matrices to save on loading time. To generate these, for a 
# given algorithm (alg), and a pre-calculated graph file (gg), again using AnNet:

# conmat <- makeConsensusMatrix(gg, N=100,alg = alg, type = 2, mask = 10,reclust = FALSE, Cnmin = -1, Cnmax = 10)



GenerateFig3c <- function(loc, clust){
  gg <- get(paste(loc, "cluster", sep="_"))
  conmat <- readRDS(paste("matrices/", clust, "_" , loc, "_matrix.rds", sep = ""))
  bm <- getBridgeness(gg,clust,conmat)
  retList <- list("graph" = gg, "matrix" = conmat, "bridge" = bm)
  return(retList)
}

locShorttoLong <- function(loc){
    return(switch(loc,
                  "post" = "Postsynaptic",
                  "pre" = "Presynaptic",
                  "syn" = "Synaptosome"))
}

centralityInput <- reactive({
        switch(input$Centrality1,
               "Semi-Local Centrality (SL)" = "SL",
               "Degree Centrality (DEG)" = "DEG",
               "Betweenness Centrality (BET)" = "BET",
               "Closeness Centrality (CC)" = "CC",
               "mean Shortest Path Centrality (mnSP)" = "mnSP",
               "Page Rank Centrality (PR)" = "PR",
               "Single-destination Shortest Path Centrality (sdSP)" = "sdSP"
        )
    })

# this function has been put in a separate reactive statement so that changing centralities or highlighting
# does not require a full recalculation of bridgeness
getdata3c <- reactive({
  print("Calculating 3c")
  if (exists("data3c")){
      rm(data2c)
  }
  data3c <- GenerateFig3c(input$LocalisationnoAll1, tolower(input$Cluster1))
})
    
    
fig3c <- eventReactive(input$run3c, {
  if (exists("data3c")){
      rm(data3c)
  }
  data3c <- getdata3c()
  print("3c calculated")
  gg <- data3c$graph
  bm <- data3c$bridge
  maxPapers <- switch(input$LocalisationnoAll1,
                      "post" = 29,
                      "pre" = 18,
                      "syn" = 11)
  minPapers <- maxPapers - as.numeric(input$vips)
  VIP_candidates <- GetNumPapersSubset(locShorttoLong(input$LocalisationnoAll1), "All", "All") %>%
    select(HumanEntrez, NumPapers) %>% unique() %>% filter(NumPapers >= minPapers)
  VIPs <- VIP_candidates$HumanEntrez
  indx <- match(V(gg)$name,VIPs)
  group <- ifelse(is.na(indx), 0,1)
  MainDivSize <- 0.8
  Xlab <- input$Centrality1
  Ylab <- "Bridgeness (B)"
  X <- as.numeric(igraph::get.vertex.attribute(gg,centralityInput(),V(gg)))
  X <- scale(X)
  Y <- as.numeric(as.vector(bm[,dim(bm)[2]]))
  lbls <- ifelse(!is.na(indx),V(gg)$GeneName,"")
  dt<-data.frame(X=X,Y=Y,vips=group,entres=V(gg)$name,name=V(gg)$GeneName)
  dt_vips <-dt[dt$vips==1,]
  dt_res <- dt[dt$vips==0,]
  baseColor<-"royalblue2"
  
  ggplot(dt,aes(x=X,y=Y,label=name))+
      geom_point(data=dt_vips,
                 aes(x=X,y=Y),colour=baseColor,size = 7,shape=15,show.legend=F)+
      geom_point(data=dt_res,
                 aes(x=X,y=Y, alpha=(X*Y)), size = 3,shape=16,show.legend=F)+
      geom_label_repel(aes(label=as.vector(lbls)),
                       fontface='bold',color='black',fill='white',box.padding=0.1,
                       point.padding=NA,label.padding=0.15,segment.color='black',
                       force=1,size=rel(3.8),show.legend=F,max.overlaps=1000)+
      scale_x_continuous(expand = c(0, 0), limits = c(0, 1)) +
      scale_y_continuous(expand = c(0, 0), limits = c(0, 1))+
      xlab(Xlab) + ylab(Ylab) +
      theme(
          axis.title.x=element_text(face="bold",size=rel(2.5)),
          axis.title.y=element_text(face="bold",size=rel(2.5)),
          legend.title=element_text(face="bold",size=rel(1.5)),
          legend.text=element_text(face="bold",size=rel(1.5)),
          legend.key=element_blank())+
      theme(panel.grid.major = element_line(colour="grey40",size=0.2),
            panel.grid.minor = element_line(colour="grey40",size=0.1),
            panel.background = element_rect(fill="white"),
            panel.border = element_rect(linetype="solid",fill=NA))+
      geom_vline(xintercept=0.5,colour="grey40",size=MainDivSize,linetype=2,show.legend=F)+
      geom_hline(yintercept=0.5,colour="grey40",size=MainDivSize,linetype=2,show.legend=F)
})
```


```{r 3coutput, echo=FALSE}
output$download3c <- downloadHandler(
  filename = function(){
    if (input$download3c == "3c(g)"){"data3c(g).gml"}
    else if (input$download3c == "3c(m)"){"data3c(m).RDS"}
    else if (input$download3c == "3c(b)"){"data3c(b).csv"}
  },
  content = function(file){
    if (input$download3c == "3c(g)"){
    write_graph(getdata3c()$graph, file, format=c("gml"))
    }
    else if (input$download3c == "3c(m)"){
      saveRDS(getdata3c()$matrix, file)
    }
    else if (input$download3c == "3c(b)"){
      write.csv(getdata3c()$bridge, file, row.names = FALSE)
    }
  }
  )

inputPanel(
  selectLocalisationNoAll(1),
  selectClustering(1),
  selectInput(
      inputId = "Centrality1",
      label = "Centrality measure",
      choices = c("Semi-Local Centrality (SL)", "Degree Centrality (DEG)", "Betweenness Centrality (BET)", "Closeness Centrality (CC)", "mean Shortest Path Centrality (mnSP)", "Page Rank Centrality (PR)", "Single-destination Shortest Path Centrality (sdSP)"),
      selected = "SL",
      multiple = FALSE
  ),
  selectInput(
      inputId = "vips",
      label = "Highlight core genes (seen in all but n papers)",
      choices = c(0, 1, 2, 3, 4, 5),
      selected = 1,
      multiple = FALSE
  ),
  # click to run the figure
  actionButton("run3c", "Click to execute figure 3c"),
  selectInput(
    inputId = "download3c",
    label = "Choose download",
    choices = c("Graph" = "3c(g)", "Consensus Matrix" = "3c(m)", "Bridgeness" = "3c(b)"),
    selected = "Graph",
    multiple = FALSE
  ),
  downloadButton("download3c", "Download current dataset")
)

renderPlot(fig3c(), height=600)

```


```{r title3d, echo=FALSE}
h3("Fig 3d:  Correlation plot for diferent centrality measures \n estimated for selected network")
```


```{r fig3d}
GenerateFig3d <- function(loc){
  gg <- get(paste(loc, "cluster", sep="_"))
  num_rows <- length(as.numeric(igraph::get.vertex.attribute(gg,"sdSP",V(gg))))
  cent_measure <- c("SL", "DEG", "BET", "CC", "mnSP", "PR", "sdSP")
  
  centrality_df <- data.frame(matrix(nrow=num_rows, ncol=length(cent_measure)))
  for (i in 1:length(cent_measure)){
      cent <- as.numeric(igraph::get.vertex.attribute(gg,cent_measure[i],V(gg)))
      centrality_df[i] <- cent
  }
  colnames(centrality_df) <- cent_measure
  cor_mat <- cor(centrality_df, use="complete.obs")
  
  return(cor_mat)
}

    
fig3d <- reactive({
  data3d <- GenerateFig3d(input$LocalisationnoAll2)
  corrplot(data3d, type="upper", tl.col = "black", tl.srt = 45, tl.cex = 1,
           col = COL2('PuOr', 10))
    })
```


```{r 3doutput, echo=FALSE}
output$download3d <- downloadHandler(
  filename = function(){"data3d.csv"},
  content = function(file){write.csv(GenerateFig3d(input$LocalisationnoAll2), file, row.names = TRUE)}
)

inputPanel(
  selectLocalisationNoAll(2),
  downloadButton("download3d", "Download current dataset")
)

renderPlot(fig3d())

```


```{r title3e, echo=FALSE}
h3("Fig 3e: Disease-disease relationship for selected interactome(s). Where significance q-values < selected, is delineated by the dashed line.")
```


```{r fig3e}
# Again calculating disease pair q-values takes a long time, so we have saved these files instead of recalculating from scratch.

# To recalculate disease interactions, for a given cluster graph (gg), use the following command from AnNet
# (the example command given here includes the diseases AD, BD, AUT, and Epi):

# r <- runPermDisease(
#  gg,
#  name = "TopOnto_OVG_HDO_ID",
#  diseases = c("DOID:10652","DOID:3312","DOID:12849","DOID:1826"),
#  Nperm = 100,
#  alpha = c(0.05, 0.01, 0.001)
# )

MapIDtoDisease <- function(id){
    ids <- c("DOID:10652", "DOID:3312", "DOID:12849", "DOID:5419",
             "DOID:0060041", "DOID:1826", "DOID:1059", "DOID:10763",
             "DOID:12858", "DOID:14330", "DOID:9255", "DOID:2377")
    dis <- c("AD", "BD", "AUT", "SC", "ASD", "Epi", "ID", "HTN",
             "HD", "PD", "FTD", "MS")
    indx <- match(c(id), ids)
    return(dis[indx])
}


GenerateFig3e <- function(loc, dis){
    disease_table <- data.frame()
    if ("post" %in% loc){
        post_table <- data.frame(readRDS("disease_pairs/post_disease_pairs.RDS")) %>%
          mutate(Localisation = "Postsynaptic")
        disease_table <- rbind(disease_table, post_table)
    }
    if ("pre" %in% loc){
        pre_table <- data.frame(readRDS("disease_pairs/pre_disease_pairs.RDS")) %>%
          mutate(Localisation = "Presynaptic")
        disease_table <- rbind(disease_table, pre_table)
    }
    if ("syn" %in% loc){
        syn_table <- data.frame(readRDS("disease_pairs/syn_disease_pairs.RDS")) %>%
          mutate(Localisation = "Synaptosome")
        disease_table <- rbind(disease_table, syn_table)
    }
    colnames(disease_table)[3] <- "HDO.ID2"
    disease_table <- disease_table %>% filter(HDO.ID != HDO.ID2)
    for (i in 1:length(disease_table$HDO.ID)){
        disease_table$HDO.Name <- MapIDtoDisease(as.character(disease_table$HDO.ID))
        disease_table$HDO2.Name <- MapIDtoDisease(as.character(disease_table$HDO.ID2))
    }
    disease_table <- disease_table %>% filter(HDO.Name %in% dis) %>% filter(HDO2.Name %in% dis)  %>% 
        mutate(Disease.Pairs = paste(HDO.Name, HDO2.Name, sep="-")) %>% 
      select(Disease.Pairs, q.value, Localisation) %>% mutate(q.value = as.numeric(as.character(q.value))) %>%
      mutate(log.qval = -log10(q.value)) %>% filter(Disease.Pairs != 'ASD-AUT')
    return(disease_table)
    
}

    
fig3e <- reactive({
        data3e <- GenerateFig3e(input$LocalisationnoAll3, input$Diseases1)
        ggplot(data3e, aes(x=log.qval, y=reorder(Disease.Pairs, log.qval, input$order), colour=Localisation)) + 
          geom_point(size=5) + xlab("-log10(q-value)") + ylab("Disease Pairs") + 
          theme(axis.title=element_text(size=14, face="bold")) + 
          geom_vline(xintercept=-log10(input$qval), linetype=2) + xlim(NA, input$xlimit)
    })
```


```{r 3eoutput, echo=FALSE}
output$download3e <- downloadHandler(
  filename = function(){"data3e.csv"},
  content = function(file){write.csv(GenerateFig3e(input$LocalisationnoAll3, input$Diseases1), file, row.names = FALSE)}
)

inputPanel(
  selectLocalisationNoAll(3, TRUE, c("post", "pre")),
  selectInput(
      inputId = "Diseases1",
      label = "Diseases",
      choices = c("AD", "BD", "AUT", "SC", "ASD", "Epi", "ID", "HTN", "HD", "PD", "FTD", "MS"),
      selected = c("AD", "BD", "SC", "ASD", "Epi", "ID", "HTN",
                   "HD", "PD", "FTD", "MS"),
      multiple = TRUE
  ),
  numericInput(
      inputId = "xlimit",
      label = "Upper x limit",
      value = 60,
      min = 1
  ),
  numericInput(
      inputId = "qval",
      label = "Q-value significance threshold",
      value = 0.05,
      min = 1e-100,
      max = 1
  ),
  selectInput(
      inputId = "order",
      label = "Sort by q-value",
      choices = c('mean', 'median', 'max', 'min'),
      selected = 'mean',
      multiple = FALSE
  ),
  downloadButton("download3e", "Download current dataset")
)

renderUI(renderPlot(fig3e(), height = 80*length(input$Diseases1)))

```



```{r title3f, echo=FALSE}
h3("Fig 3f: Colocalization of chosen disease on selected network")
```


```{r fig3f}
GenerateFig3f <- function(loc, clust, dis){
  gg <- get(paste(loc, "cluster", sep="_"))
  if (dis == "None"){
      mem.df<-data.frame(GeneName=V(gg)$GeneName,membership=as.numeric(vertex_attr(gg, clust)))
  }
  else{
      hasDisease <- function(x){grepl(dis, x)}
      mem.df<-data.frame(GeneName=V(gg)$GeneName,membership=as.numeric(vertex_attr(gg, clust)), 
                         disease=sapply(V(gg)$TopOntoOVG, hasDisease))
  }
  return(list("graph" = gg, "mem.df" = mem.df))
}

getdata3f <- reactive({
        data3f <- GenerateFig3f(input$LocalisationnoAll4, tolower(input$Cluster2), input$Diseases2)
    })
    
layout <- eventReactive({input$LocalisationnoAll4
  input$Cluster2}, {
    # the layoutByCluster function is taken from the package AnNet
    lay<-layoutByCluster(getdata3f()$graph,getdata3f()$mem.df,layout = layout_nicely)
  })
    
fig3f <- reactive({
  data3f <- getdata3f()
  gg <- data3f$graph
  mem.df <- data3f$mem.df
  
  palette <- distinctColorPalette(max(as.numeric(mem.df$membership)))
  lay<-layout()
  mem.df$x <- lay[,1]
  mem.df$y <- lay[,2]
  if (input$Diseases2 == "None"){
      mem.df$fill <- palette[as.numeric(mem.df$membership)]
      mem.df$disease <- "None"
  }
  else{
      mem.df$fill <- c("orange", "lightblue")[1+as.numeric(mem.df$disease)]
  }
  static_3f <- ggplot(mem.df, aes(x=x, y=y, fill=fill, GeneName=GeneName, Membership=membership, 
                                  Disease=disease)) +  geom_point() + 
    theme(axis.line=element_blank(),axis.text.x=element_blank(),
          axis.text.y=element_blank(),axis.ticks=element_blank(),
          axis.title.x=element_blank(),axis.title.y=element_blank(),legend.position="none",
          panel.background=element_blank(),panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),plot.background=element_blank(),
          panel.border = element_rect(colour = "black", fill=NA, size=1))
  
  ggplotly(static_3f,tooltip = c("GeneName", "Membership", "Disease"))
})
```


```{r 3foutput, echo=FALSE}
output$download3f <- downloadHandler(
  filename = function(){
    if (input$down3f == "Figure"){"data3f.csv"}
    else if (input$down3f == "Table"){"dataTable3f.csv"}
  },
  content = function(file){
    if (input$down3f == "Figure"){
      write.csv(GenerateFig3f(input$LocalisationnoAll3, tolower(input$Cluster2), input$Diseases2)$mem.df, file, row.names = FALSE)
    }
    else if (input$down3f == "Table"){
     write.csv(GenerateTab3f(getdata3f()$mem.df), file, row.names = FALSE)
    }
  }
)

inputPanel(
  selectLocalisationNoAll(4),
  selectClustering(2),
  selectInput(
    inputId = "Diseases2",
    label = "Disease",
    choices = c("None (highlight clusters)" = "None", "AD", "BD", "AUT", "SC", "ASD", "Epi", "ID", "HTN",
                "HD", "PD", "FTD", "MS"),
    selected = "None",
    multiple = FALSE
  ),
  selectInput(
    inputId = "down3f",
    label = "Choose download",
    choices = c("Figure (proteins with clusters and disease association)" = "Figure","Table (p-value by cluster)" = "Table"),
    selected = "Graph",
    multiple = FALSE
  ),
  downloadButton("download3f", "Download current dataset")
)

renderPlotly(fig3f())#, height=600, width=600)

```


```{r title3ftab, echo=FALSE}
h3("Table 3f: Overrepresentation analysis for chosen disease for each cluster.\n p-values < 0.01 highlighted in green, p-values < 0.05 highlighted in blue.", align='center')
```


```{r tab3f}
GenerateTab3f <- function(mem.df){
  if ("disease" %in% colnames(mem.df)){
      b <- sum(mem.df$disease)
      c <- length(mem.df$disease)
      pval.df <-mem.df %>% group_by(membership) %>% summarise(num.disease.genes = sum(disease), 
                                                              num.total.genes = length(disease)) %>% 
        mutate(b = b, c = c)
      pval <- c()
      for (i in 1:length(pval.df$membership)){
          pval <- append(pval, min(1-cumsum(dhyper(0:(pval.df$num.disease.genes[i]-1),
                                                   pval.df$b[i],(pval.df$c[i]-pval.df$b[i]),
                                                   pval.df$num.total.genes[i]))))
      }
      pval.df <- pval.df %>% mutate(p.value = pval) %>% select(-c(c, b))
  }
  else{
      pval.df <- data.frame("Disease" = "No disease selected")
  }
  return(pval.df)
}


diseaseTable <- reactive({
  if (input$Diseases2 != "None"){
      mem.df <- getdata3f()$mem.df
      pval.df <- GenerateTab3f(mem.df)
      
      datatable(pval.df, rownames = FALSE) %>%
          formatStyle('p.value', target = 'row', backgroundColor = styleInterval(c(0.01, 0.05),                                                        c('lightgreen', "lightblue", "white")))
  }
  else{
      datatable(data.frame("Disease" = "No disease selected"), rownames = FALSE) 
  }
})
```


```{r 3ftaboutput, echo=FALSE}
renderDataTable(diseaseTable())

```

```{r finaltext, echo=FALSE}
h4('There are many complex co-morbidities between psychiatric disorders at the population and the genetic level but for most the molecular basis remains elusive. The network perspective can be used to obtain a different view by linking topology and phenotype together. Gene-disease association data is noisy and far from complete, but we can partly compensate by measuring, for each disease, the distance from each protein in the network to its nearest known associated protein, which can be extended to disease pairs [66] to dissect how these different neurological diseases coalesce at the synapse.')
        
h4('Using PSP (both full and consensus) and presynaptic networks we found clear evidence of network overlap between well-known co-morbid neuro-psychiatric/developmental disorders in both postsynaptic and presynaptic models (q-values shown for PSP/presynaptic networks), including BD-SCH (P = 2.0E-49/4.39E-16), BD-ASD (P = 7.12E-20/1.28E-7), and ASD/SCH (P = 6.17E-16/1.12E-5). Similarly, overlap was observed for common neurodegenerative diseases/conditions AD and PD (P = 3.04E-6/1.32E-6).')
        
h4(HTML('We also observed compartment-specific overlaps for Epilepsy with PD (P = 0.53/2.12E-3) and BD (P = 0.54/9.73E-4), which is significant only in the presynaptic network (Fig. 3E).'))
        
h4(HTML('In both postsynaptic and presynaptic models, we found overlap for Hypertension (HTN) with AD (P = 8.6E-4/1.0E-2, and with MS (P = 8.79E-5/2.12E-3) (Fig. 3E). The AD-HTN link is not, in itself, new but commonly considered as a cardiovascular mechanism with a neurological impact. However, the network view reveals a new potential mechanistic link at the synapse. Although we found significant overlaps between AD-HTN and AD-PD, we did not see evidence for a PD-HTN link (P = 0.17/0.36), which indicates the potential shared mechanistic pathway between AD and HTN, which is different to the pathways shared between AD and PD (Fig. 3E).'))
        
h4(HTML("To further dissect the potential sharing of pathways between AD and HTN in the PSP network (Fig. 3F), we employed Belief Propagation to propagate these GDA's through the network's edges, and a Degree-Corrected Block Model (DC-SBM) to model its effect on network clustering [67]. Under a prior assumption of no correlation between the GDA's and the network communities, we found evidence for the co-localization of AD and HTN (C = 31 P = 4.69E-5 and C = 43 P = 1.6E-11). Functionally, these communities are enriched for synaptic transmission, axon guidance (C = 31, GO:0007268 = 5.8E-3, GO:0007411 = 7.46E-5), stress activated MAPK cascade and response to oxidative stress (C = 43, GO:0051403 = 1.92E-5, GO:0006979 = 5.34E-5)."))
        
h4('The presented synapse proteome dataset is the largest, most complete and up to date and is freely available with lightweight tools to allow anyone to extract relevant subsets. It compliments previously published curated dataset of synaptic genes SynGO [68], and both resources could be used jointly as we have cross-referenced the common genes. By mirroring the methods used it would be straightforward for any user to add in their own datasets for comparison.')

h1("References")
        
tags$ol(tags$li("Husi, H. et al. Proteomic analysis of NMDA receptor-adhesion protein signaling complexes. Nat. Neurosci. 3(7), 661-669 (2000)."), 
tags$li("Walikonis, R. S. et al. Identifcation of proteins in the postsynaptic density fraction by mass spectrometry. J. Neurosci. 20(11), 4069-4080 (2000)."),
  tags$li("Peng, J. et al. Semiquantitative proteomic analysis of rat forebrain postsynaptic density fractions by mass spectrometry. J. Biol. Chem. 279(20), 21003-21011 (2004)"),
  tags$li("Satoh, K. et al. Identification of activity-regulated proteins in the postsynaptic density fraction. Genes Cells 7(2), 187-197 (2002)."),
  tags$li("Yoshimura, Y. et al. Molecular constituents of the postsynaptic density fraction revealed by proteomic analysis using multidimensional liquid chromatography-tandem mass 
      spectrometry. J. Neurochem. 88(3), 759-768 (2004)."),
  tags$li("Farr, C. D. et al. Proteomic analysis of native metabotropic glutamate receptor 5 protein complexes reveals novel molecular constituents. J.
      Neurochem. 91(2), 438-450 (2004)."),
  tags$li("Jordan, B. A. et al. Identification and verifcation of novel rodent postsynaptic density proteins. Mol. Cell Proteom. 3(9), 857-871(2004)."),
  tags$li("Li, K. W. et al. Proteomics analysis of rat brain postsynaptic density. Implications of the diverse protein functional groups for the integration of synaptic physiology.
      J. Biol. Chem. 279(2), 987-1002 (2004)."),
  tags$li("Trinidad, J. C. et al. Phosphorylation state of postsynaptic density proteins. J. Neurochem. 92(6), 1306-1316 (2005)."),
  tags$li("Cheng, D. et al. Relative and absolute quantifcation of postsynaptic density proteome isolated from rat forebrain and cerebellum. Mol. Cell
      Proteom. 5(6), 1158-1170 (2006)."),
  tags$li("Collins, M. O. et al. Molecular characterization and comparison of the components and multiprotein complexes in the postsynaptic proteome. J. Neurochem. 
      97(Suppl 1), 16-23 (2006)."),
  tags$li("Dosemeci, A. et al. Preparation of postsynaptic density fraction from hippocampal slices and proteomic analysis. Biochem. Biophys. Res. Commun. 339(2), 687-694 (2006)."),
  tags$li("Dosemeci, A. et al. Composition of the synaptic PSD-95 complex. Mol. Cell Proteom. 6(10), 1749-1760 (2007)."),
  tags$li("Trinidad, J. C. et al. Quantitative analysis of synaptic phosphorylation and protein expression. Mol. Cell Proteom. 7(4), 684-696 (2008)."),
  tags$li("Selimi, F. et al. Proteomic studies of a single CNS synapse type: The parallel fiber/purkinje cell synapse. PLoS Biol. 7(4), e83 (2009)."),
  tags$li("Fernandez, E. et al. Targeted tandem affinity purification of PSD-95 recovers core postsynaptic complexes and schizophrenia susceptibility proteins. 
      Mol. Syst. Biol. 5, 269 (2009)."),
  tags$li("Bayes, A. et al. Characterization of the proteome, diseases and evolution of the human postsynaptic density. Nat. Neurosci. 14(1), 19-21 (2011)."),
  tags$li("Bayes, A. et al. Comparative study of human and mouse postsynaptic proteomes fnds high compositional conservation and abundance differences for key synaptic proteins. 
      PLoS ONE 7(10), e46683 (2012)."),
  tags$li("Schwenk, J. et al. High-resolution proteomics unravel architecture and molecular diversity of native AMPA receptor complexes. Neuron 74(4), 621-633 (2012)."),
  tags$li("Distler, U. et al. In-depth protein profiling of the postsynaptic density from mouse hippocampus using data-independent acquisition proteomics. 
      Proteomics 14(21-22), 2607-2613 (2014)."),
  tags$li("Bayes, A. et al. Human post-mortem synapse proteome integrity screening for proteomic studies of postsynaptic complexes. Mol. Brain 7, 88 (2014)."),
  tags$li("Seo, T. K. & Torne, J. L. Information criteria for comparing partition schemes. Syst. Biol. 67(4), 616-632 (2018)."),
  tags$li("Focking, M. et al. Proteomic analysis of the postsynaptic density implicates synaptic function and energy pathways in bipolar disorder. Transl. Psychiatry 6(11), e959 
      (2016)."),
  tags$li("Li, J. et al. Long-term potentiation modulates synaptic phosphorylation networks and reshapes the structure of the postsynaptic interactome. Sci. Signal.
      9(440), 8 (2016)."),
  tags$li("Fernandez, E. et al. Arc requires PSD95 for assembly into postsynaptic complexes involved with neural dysfunction and intelligence. Cell Rep. 21(3), 679-691 (2017)."),
  tags$li("Roy, M. et al. Proteomic analysis of postsynaptic proteins in regions of the human neocortex. Nat. Neurosci. 21(1), 130-138 (2018)."),
  tags$li("Li, J. et al. Spatiotemporal profle of postsynaptic interactomes integrates components of complex brain disorders. Nat. Neurosci. 20(8), 1150-1161 (2017)."),
  tags$li("Roy, M. et al. Regional diversity in the postsynaptic proteome of the mouse brain. Proteomes 6, 3 (2018)."),
  tags$li("Wilson, R. S. et al. Development of targeted mass spectrometry-based approaches for quantitation of proteins enriched in the postsynaptic density
      (PSD). Proteomes 7, 2 (2019)."),
  tags$li("Coughenour, H. D., Spaulding, R. S. & Tompson, C. M. The synaptic vesicle proteome: A comparative study in membrane protein identification. Proteomics 4(10),
      3141-3155 (2004)."),
  tags$li("Blondeau, F. et al. Tandem MS analysis of brain clathrin-coated vesicles reveals their critical involvement in synaptic vesicle recycling. Proc. Natl. Acad. Sci.
      U.S.A. 101(11), 3833-3838 (2004)."),
  tags$li("Phillips, G. R. et al. Proteomic comparison of two fractions derived from the transsynaptic scafold. J. Neurosci. Res. 81(6), 762-775 (2005)."),
  tags$li("Morciano, M. et al. Immunoisolation of two synaptic vesicle pools from synaptosomes: A proteomics analysis. J. Neurochem. 95(6), 1732-1745 (2005)."),
  tags$li("Burre, J. et al. Analysis of the synaptic vesicle proteome using three gel-based protein separation techniques. Proteomics 6(23), 6250-6262 (2006)."),
  tags$li("Takamori, S. et al. Molecular anatomy of a trafficking organelle. Cell 127(4), 831-846 (2006)."),
  tags$li("Khanna, R., Zougman, A. & Stanley, E. F. A proteomic screen for presynaptic terminal N-type calcium channel (CaV2.2) binding partners. J. Biochem. Mol. 
      Biol. 40(3), 302-314 (2007)."),
  tags$li("Morciano, M. et al. The proteome of the presynaptic active zone: From docked synaptic vesicles to adhesion molecules and maxichannels. J. 
      Neurochem. 108(3), 662-675 (2009)."),
  tags$li("Abul-Husn, N. S. et al. Systems approach to explore components and interactions in the presynapse. Proteomics 9(12), 3303-3315 (2009)."),
  tags$li("Abul-Husn, N. S. et al. Chronic morphine alters the presynaptic protein profile: Identification of novel molecular targets using proteomics and network 
      analysis. PLoS ONE 6(10), e25535 (2011)."),
  tags$li("Gorini, G. et al. Dynamin-1 co-associates with native mouse brain BKCa channels: Proteomics analysis of synaptic protein complexes. FEBS Lett. 584(5), 845-851 (2010)."),
  tags$li("Gronborg, M. et al. Quantitative comparison of glutamatergic and GABAergic synaptic vesicles unveils selectivity for few proteins including MAL2, a novel synaptic vesicle 
      protein. J. Neurosci. 30(1), 2-12 (2010)."),
  tags$li("Boyken, J. et al. Molecular profling of synaptic vesicle docking sites reveals novel proteins but few diferences between glutamatergic and GABAergic synapses. 
      Neuron 78(2), 285-297 (2013)."),
  tags$li("Wilhelm, B. G. et al. Composition of isolated synaptic boutons reveals the amounts of vesicle trafficking proteins. Science 344(6187), 1023-1028 (2014)."),
  tags$li("Brinkmalm, A. et al. Targeting synaptic pathology with a novel affinity mass spectrometry approach. Mol. Cell Proteom. 13(10), 2584-2592 (2014)."),
  tags$li("Weingarten, J. et al. The proteome of the presynaptic active zone from mouse brain. Mol. Cell Neurosci. 59, 106-118 (2014)."),
  tags$li("Kokotos, A. C. et al. Activity-dependent bulk endocytosis proteome reveals a key presynaptic role for the monomeric GTPase Rab11. Proc. Natl. Acad. Sci.
      U.S.A. 115(43), E10177-e10186 (2018)."),
  tags$li("Taoufq, Z. et al. Hidden proteome of synaptic vesicles in the mammalian brain. Proc. Natl. Acad. Sci. U.S.A. 117(52), 33586-33596 (2020)"),
  tags$li("Filiou, M. D. et al. Profiling of mouse synaptosome proteome and phosphoproteome by IEF. Electrophoresis 31(8), 1294-1301 (2010)."),
  tags$li("Dahlhaus, M. et al. The synaptic proteome during development and plasticity of the mouse visual cortex. Mol. Cell Proteom. 10(5),5413 (2011)."),
  tags$li("Cohen, L. D. et al. Metabolic turnover of synaptic proteins: Kinetics, interdependencies and implications for synaptic maintenance. PLoS ONE 8(5), e63191 (2013)."),
  tags$li("Biesemann, C. et al. Proteomic screening of glutamatergic mouse brain synaptosomes isolated by fuorescence activated sorting. Embo J. 33(2), 157-170 (2014)."),
  tags$li("Chang, R. Y. et al. SWATH analysis of the synaptic proteome in Alzheimer's disease. Neurochem. Int. 87, 1-12 (2015)."),
  tags$li("Liu, X. A. et al. New approach to capture and characterize synaptic proteome. Proc. Natl. Acad. Sci. U.S.A. 111(45), 16154-16159 (2014)."),
  tags$li("Kohansal-Nodehi, M. et al. Analysis of protein phosphorylation in nerve terminal reveals extensive changes in active zone proteins upon exocytosis. Elife 5, 2 (2016)."),
  tags$li("Gonzalez-Lozano, M. A. et al. Dynamics of the mouse brain cortical synaptic proteome during postnatal brain development. Sci. Rep. 6, 35456 (2016)."),
  tags$li("Alfieri, A. et al. Synaptic interactome mining reveals p140Cap as a new hub for PSD proteins involved in psychiatric and neurological disorders. Front. Mol. 
      Neurosci. 10, 212 (2017)."),
  tags$li("Heo, S. et al. Identification of long-lived synaptic proteins by proteomic analysis of synaptosome protein turnover. Proc. Natl. Acad. Sci. U.S.A. 115(16), 
      E3827-e3836 (2018)."),
  tags$li("Oughtred, R. et al. The BioGRID interaction database: 2019 update. Nucleic Acids Res. 47(D1), D529-541 (2019)."),
  tags$li("Kerrien, S. et al. The IntAct molecular interaction database in 2012. Nucleic Acids Res. 40, 841-846 (2012)."),
  tags$li("Xenarios, I. et al. DIP, the database of interacting proteins: A research tool for studying cellular networks of protein interactions. Nucleic Acids Res.
      30(1), 303-305 (2002)."),
  tags$li("Newman, M. E. Modularity and community structure in networks. Proc. Natl. Acad. Sci. U.S.A. 103(23), 8577-8582 (2006)."),
  tags$li("McLean, C. et al. Improved functional enrichment analysis of biological networks using scalable modularity based clustering. J. Proteom. Bioinform. 9(1), 9-18 (2016)."),
  tags$li("Han, J. D. et al. Evidence for dynamically organized modularity in the yeast protein-protein interaction network. Nature 430(6995), 88-93 (2004)."),
  tags$li("Nepusz, T., Yu, H. & Paccanaro, A. Detecting overlapping protein complexes in protein-protein interaction networks. Nat. Methods 9(5), 471-472 (2012)."),
  tags$li("Teschendorf, A. E. et al. Increased signaling entropy in cancer requires the scale-free property of protein interaction networks. Sci. Rep. 5, 9646 (2015)."),
  tags$li("Menche, J. et al. Disease networks. Uncovering disease-disease relationships through the incomplete interactome. Science 347(6224), 1257601 (2015)."),
  tags$li("Newman, M. E. & Clauset, A. Structure and inference in annotated networks. Nat. Commun. 7, 11863 (2016)."),
  tags$li("Koopmans, F. et al. SynGO: An evidence-based, expert-curated knowledge base for the synapse. Neuron 103(2), 217-234.e4 (2019)."))
        
```
